tests/test_helpers.py

```
@@ -56,6 +65,29 @@ def test_coerce_bool_returns_none_for_unknown_values(value: Any) -> None:
+def test_is_usable_carp_vip_accepts_normalized_identity_without_interface(
+    value: dict[str, Any],
+) -> None:
+    """CARP VIP usability should accept integer/string VHIDs without interface names."""
+    assert is_usable_carp_vip(value) is True
+
+
+@pytest.mark.parametrize(
+    "value",
+    [None, {}, [], {"vhid": "", "subnet": "192.0.2.1"}, {"vhid": 1, "subnet": ""}],
+)
+def test_is_usable_carp_vip_rejects_missing_or_blank_identity(value: Any) -> None:
```

Ruff (0.15.20)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)